elisp/guix-env-var.el

```diff
@@ -0,0 +1,98 @@
+;;; guix-env-var.el --- Guix environment variables mode
+
+;; Copyright © 2018 Oleg Pykhalov <go.wigust@gmail.com>
+
+;; This file is part of Emacs-Guix.
+
+;; Emacs-Guix is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; Emacs-Guix is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with Emacs-Guix.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; This file provides `guix-env-var-mode' the major mode for Guix
+;; to prettify environment-variables and profile files.
+
+;;; Code:
+
+(defgroup guix-env-var nil
+  "Settings for `guix-env-var-mode'."
+  :prefix "guix-env-var-"
+  :group  'guix)
+
+(defcustom guix-env-var-enable-formatting t
+  "If non-nil, prettify Guix environment-variables and profile files."
+  :type 'boolean
+  :group 'guix-env-var)
+
+(defvar guix-env-var-display-message t
+  "If non-nil, display a message after enabling `guix-env-var-mode'.
+This message allows you to see that the current buffer is
+indented (while the original buffer file may not be indented).")
+
+(defvar guix-env-var-regexp
+  (rx "export" (one-or-more space)
+      letter (zero-or-more (or alphanumeric "_")) "=")
+  "Regexp matching shell variables in Guix environment-variables file.")
+
+(defvar guix-env-var-comment-line-regexp
+  (rx (zero-or-more whitespace) "#")
+  "Regexp matching a line beginning with a comment symbol.")
+
+(defun guix-env-var-comment-line-p ()
+  "Return non-nil if current line begins with a comment."
+  (save-excursion
+    (beginning-of-line)
+    (re-search-forward guix-env-var-comment-line-regexp (line-end-position) t)))
+
+;;;###autoload
+(defun guix-env-var-prettify-variable ()
+  "Prettify Shell environment variable at current line."
+  (interactive)
+  (let ((env-beg (line-beginning-position))
+        (env-end (line-end-position)))
+    (when (not (guix-env-var-comment-line-p))
+      (narrow-to-region env-beg env-end)
+      (beginning-of-line)
+      (when (search-forward "=" nil t)
+        (insert "\\\n")
+        (while (search-forward ":/" nil t)
+          (backward-char 2)
+          (insert "\\\n")
+          (forward-char)))
+      (end-of-line)
+      (widen))))
+
+;;;###autoload
+(defun guix-env-var-prettify-buffer (&optional buffer)
+  "Prettify BUFFER with `guix-env-var-prettify-variable'."
+  (interactive)
+  (let ((buffer (unless buffer (current-buffer))))
```

Comment:
You don't use this `buffer` variable, so I added `with-current-buffer` here (I think that's what you meant).  Also I added `save-excursion` to avoid moving to the end of buffer after it is formatted.